lib/miq_automation_engine/engine/miq_ae_engine/miq_ae_object.rb

```diff
@@ -557,11 +557,11 @@ def self.convert_value_based_on_datatype(value, datatype)
       return convert_boolean_value(value)                    if datatype == 'boolean'
       return true                                            if datatype == 'TrueClass'
       return false                                           if datatype == 'FalseClass'
-      return Time.parse(value)                               if datatype == 'time' || datatype == 'Time'
-      return value.to_sym                                    if datatype == 'symbol' || datatype == 'Symbol'
-      return value.to_i                                      if datatype == 'integer' || datatype == 'Fixnum'
-      return value.to_f                                      if datatype == 'float' || datatype == 'Float'
-      return value.gsub(/[\[\]]/, '').strip.split(/\s*,\s*/)  if datatype == 'array' && value.class == String
+      return Time.parse(value)                               if datatype.casecmp?('time')
+      return value.to_sym                                    if datatype.casecmp?('symbol')
+      return value.to_i                                      if datatype.casecmp?('integer') || datatype == 'Fixnum'
```

Comment:
The lower-case 'integer'  type comes from the Automate class schema which is why we initially had that value and `Fixnum`.  Ruby 2.4 replaced `Fixnum` with `Integer` which causes the issue.

Since we still support Ruby 2.3 we will need to retain the `Fixnum` check here for a bit longer.